assets/js/common/DisabledGuard/DisabledGuard.jsx

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+
+import Tooltip from '@common/Tooltip';
+
+const ALL_PERMITTED = ['all:all'];
+
+const DEFAULT_TOOLTIP_MESSAGE = 'You are not authorized for this action';
+
+// DisabledGuard adds the `disabled` prop to the guarded element.
+// The children element must be of one element.
+// If the coming children is wrapped with a tooltip, the authorization tooltip
+// supersedes the original tooltip in case of unauthorized access.
+
+function DisabledGuard({
+  userAbilities,
+  permitted,
+  withTooltip = true,
+  tooltipMessage = DEFAULT_TOOLTIP_MESSAGE,
+  children,
+}) {
+  const permittedFor = ALL_PERMITTED.concat(permitted);
+
+  const isPermitted = userAbilities
+    .map(({ name, resource }) => `${name}:${resource}`)
+    .some((ability) => permittedFor.includes(ability));
+
+  if (isPermitted) {
+    return children;
+  }
+
+  const guardedElement =
+    children.type === Tooltip ? children.props.children : children;
+
+  const disabledElement = React.cloneElement(guardedElement, {
+    disabled: true,
+  });
+
+  return (
+    <Tooltip
+      isEnabled={withTooltip}
+      content={tooltipMessage}
+      place="bottom"
```

Comment:
Can we land in a situation where we want to change the position of the Tooltip ?
Maybe add prop to `DisabledGuard` ? 
On the other hand it is an easy adjustment if we need this in the future, so no hard opinion on that

Reply:
I will just put it as prop and put `bottom` as default